pilot/pkg/bootstrap/server.go

```diff
@@ -882,7 +882,12 @@ func (s *Server) initRegistryEventHandlers() {
 	s.ServiceController().AppendServiceHandler(serviceHandler)
 
 	if s.configController != nil {
-		configHandler := func(_ config.Config, curr config.Config, event model.Event) {
+		configHandler := func(prev config.Config, curr config.Config, event model.Event) {
+			// For update events, trigger push only if spec has changed.
+			if event == model.EventUpdate && !equals(prev, curr) {
+				log.Debugf("skipping push for %s as spec has not changed", prev.Key())
```

Comment:
s.statusReporter.AddInProgressResource(curr)

this part might be broken since we skip it? Not 100% sure but I am thinking `istioctl wait` maybe get mad the config was never sent?

Reply:
I was not sure about this. So moved it to defer func so that it gets always executing

Reply:
BTW, I think we need to provide a way to completely disable `statusReporter`, currently it is always initiated, though disabled to write status.

Reply:
Its still used by `istioctl wait` even if write status is disabled. We could still provide a way to disable if needed though